astropy/coordinates/angles/core.py

```diff
@@ -183,7 +183,7 @@ def __new__(cls, angle, unit=None, dtype=np.inexact, copy=True, **kwargs):
             elif isiterable(angle) and not (
                 isinstance(angle, np.ndarray) and angle.dtype.kind not in "SUVO"
             ):
-                angle = [Angle(x, unit, copy=COPY_IF_NEEDED) for x in angle]
+                angle = [cls(x, unit, copy=COPY_IF_NEEDED) for x in angle]
```

Comment:
This looks like a good change, but is it related to the bug that is being fixed?

Reply:
Yes, it was needed, since this iteration over list of `str` is the reason a simple check on `str` input types can be used to catch the bug. (And of course it should have been written like this in the first place!)